services/player-service/adapter-persistence/src/main/java/com/kjeldsen/player/persistence/mongo/documents/TeamDocument.java

```diff
@@ -0,0 +1,39 @@
+package com.kjeldsen.player.persistence.mongo.documents;
+
+import com.kjeldsen.player.domain.Team;
+import com.kjeldsen.player.domain.TeamId;
+import com.kjeldsen.player.domain.TeamName;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.TypeAlias;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@Builder
+@Getter
+@Setter
+@Document(collection = "Teams")
+@TypeAlias("Team")
+public class TeamDocument {
+
+    public static TeamDocument from(Team team, String userId) {
+        return TeamDocument.builder()
+            .id(team.getId().value())
+            .name(team.getName().value())
+            .userId(userId)
+            .build();
+    }
+
+    @Id
+    private String id;
+    private String name;
+    private String userId;
+
+    public Team toDomain() {
```

Comment:
I think this is a good moment to add mapstruct to the project. We start having many toDomain and from methods that are clearly mappers. Plus we have kafka, rest and mongo adapters so we should clear define the mappers in the proper modules. What do you think?

Reply:
Ok! But I like the idea for those map just among layers, for example, `Infrastructure<->Domain`.Right now, I like the way in how we build our objects using factory methods like `from` or `of` or `generate`, provides better readability, better domain context and also allow us to do validation on them respecting Single Responsability Principle.